de.metas.handlingunits.base/src/test/java/de/metas/handlingunits/shipmentschedule/spi/impl/InOutProducerFromShipmentScheduleWithHUTest.java

```diff
@@ -0,0 +1,233 @@
+package de.metas.handlingunits.shipmentschedule.spi.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.sql.Timestamp;
+
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.test.AdempiereTestHelper;
+import org.adempiere.util.time.FixedTimeSource;
+import org.adempiere.util.time.SystemTime;
+import org.compiere.util.TimeUtil;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.metas.inout.model.I_M_InOut;
+import de.metas.inoutcandidate.model.I_M_ShipmentSchedule;
+
+/*
+ * #%L
+ * de.metas.handlingunits.base
+ * %%
+ * Copyright (C) 2017 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+public class InOutProducerFromShipmentScheduleWithHUTest
+{
+
+	@Before
+	public void init()
+	{
+		AdempiereTestHelper.get().init();
+	}
+
+	@Test
+	public void isCandidateShipmentDateFitForShipment_SameDate_Test()
+	{
+		SystemTime.setTimeSource(new FixedTimeSource(2017, 11, 10, 0, 0, 0));
```

Comment:
as i said, i'm not sure that this is about, but i notice that your fidex time source is always at 00:00 o'clock..
but in real live, since afaiu you use SystemTime.astimeStamp for your getNow() implemementation, that will rarely be the case..so I suggest to use some other times in your tests

Reply:
Done
Note that shipment schedule's delivery date is with timestamp and the shipment's movement date is date only.